libvips/foreign/spngsave.c

```diff
@@ -0,0 +1,837 @@
+		2, 256, 256 );
+
+}
+
+static void
+vips_foreign_save_spng_init( VipsForeignSaveSpng *spng )
+{
+	spng->compression = 6;
+	spng->filter = VIPS_FOREIGN_PNG_FILTER_ALL;
+	spng->Q = 100;
+	spng->dither = 1.0;
+	spng->effort = 7;
+}
+
+typedef struct _VipsForeignSaveSpngTarget {
+	VipsForeignSaveSpng parent_object;
+
+	VipsTarget *target;
+} VipsForeignSaveSpngTarget;
+
+typedef VipsForeignSaveSpngClass VipsForeignSaveSpngTargetClass;
+
+G_DEFINE_TYPE( VipsForeignSaveSpngTarget, vips_foreign_save_spng_target, 
+	vips_foreign_save_spng_get_type() );
+
+static int
+vips_foreign_save_spng_target_build( VipsObject *object )
+{
+	VipsForeignSaveSpng *spng = (VipsForeignSaveSpng *) object;
+	VipsForeignSaveSpngTarget *target = 
+		(VipsForeignSaveSpngTarget *) object;
+
+	spng->target = target->target;
+	g_object_ref( spng->target );
+
+	if( VIPS_OBJECT_CLASS( vips_foreign_save_spng_target_parent_class )->
+		build( object ) )
+		return( -1 );
+
+	return( 0 );
+}
+
+static void
+vips_foreign_save_spng_target_class_init( VipsForeignSaveSpngTargetClass *class )
+{
+	GObjectClass *gobject_class = G_OBJECT_CLASS( class );
+	VipsObjectClass *object_class = (VipsObjectClass *) class;
+
+	gobject_class->set_property = vips_object_set_property;
+	gobject_class->get_property = vips_object_get_property;
+
+	object_class->nickname = "pngsave_target";
+	object_class->description = _( "save image to target as PNG" );
+	object_class->build = vips_foreign_save_spng_target_build;
+
+	VIPS_ARG_OBJECT( class, "target", 1,
+		_( "Target" ),
+		_( "Target to save to" ),
+		VIPS_ARGUMENT_REQUIRED_INPUT, 
+		G_STRUCT_OFFSET( VipsForeignSaveSpngTarget, target ),
+		VIPS_TYPE_TARGET );
+
+}
+
+static void
+vips_foreign_save_spng_target_init( VipsForeignSaveSpngTarget *target )
+{
+}
+
+typedef struct _VipsForeignSaveSpngFile {
+	VipsForeignSaveSpng parent_object;
+
+	char *filename; 
+} VipsForeignSaveSpngFile;
+
+typedef VipsForeignSaveSpngClass VipsForeignSaveSpngFileClass;
+
+G_DEFINE_TYPE( VipsForeignSaveSpngFile, vips_foreign_save_spng_file, 
+	vips_foreign_save_spng_get_type() );
+
+static int
+vips_foreign_save_spng_file_build( VipsObject *object )
+{
+	VipsForeignSaveSpng *spng = (VipsForeignSaveSpng *) object;
+	VipsForeignSaveSpngFile *file = (VipsForeignSaveSpngFile *) object;
+
+	if( !(spng->target = vips_target_new_to_file( file->filename )) )
+		return( -1 );
+
+	if( VIPS_OBJECT_CLASS( vips_foreign_save_spng_file_parent_class )->
+		build( object ) )
+		return( -1 );
+
+	return( 0 );
+}
+
+static void
+vips_foreign_save_spng_file_class_init( VipsForeignSaveSpngFileClass *class )
+{
+	GObjectClass *gobject_class = G_OBJECT_CLASS( class );
+	VipsObjectClass *object_class = (VipsObjectClass *) class;
+
+	gobject_class->set_property = vips_object_set_property;
+	gobject_class->get_property = vips_object_get_property;
+
+	object_class->nickname = "pngsave";
+	object_class->description = _( "save image to files as PNG" );
```

Comment:
nit:
```suggestion
	object_class->description = _( "save image to file as PNG" );
```
or perhaps:
```suggestion
	object_class->description = _( "save image to PNG file" );
```
(noticed this description change in commit https://github.com/libvips/libvips/commit/57a0bd82f91ec776911f6d3aa85fbe2b06547612)